java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/Configuration.kt

```diff
@@ -161,36 +163,36 @@ private fun addWsSchema(address: String?): String? {
     }
 }
 
-private val pathSeparator = if (Platform.osFamily == OsFamily.WINDOWS) "\\" else "/"
-var drillInstallationDir: String? = null
 
 private fun agentParams(options: String): Map<String, String> {
-    logger.info { "agent options: $options" }
+    logger.debug { "agent options: $options" }
     val agentParams = asAgentParams(options)
-    logger.info { "agent parameters: $agentParams" }
-    drillInstallationDir = agentParams["drillInstallationDir"] ?: drillInstallationDir()
+    logger.debug { "agent parameters: $agentParams" }
+    val drillInstallationDir = agentParams[DRILL_INSTALLATION_DIR_PARAM] ?: drillInstallationDir() ?: "."
+    logger.debug { "drillInstallationDir: $drillInstallationDir" }
     val configPath = agentParams["configPath"]
         ?: getenv(SYSTEM_CONFIG_PATH)?.toKString()
         ?: "${drillInstallationDir}${pathSeparator}drill.properties"
-    logger.info { "config path: $configPath" }
-    val configParams = configPath.takeIf(String::isNotEmpty)
-        ?.runCatching(::readFile)
-        ?.getOrNull()
-        ?.let { asAgentParams(it, "\n", "#") }
-    logger.info { "config parameters: $configParams" }
-    val resultParams = configParams?.toMutableMap()?.also { it.putAll(agentParams) } ?: agentParams
+    logger.debug { "config path: $configPath" }
+    val configParams = configPath
+        .runCatching(::readFile)
+        .getOrNull()
+        .let { asAgentParams(it, "\n", "#") }
+    logger.debug { "config parameters: $configParams" }
+    val resultParams = configParams.toMutableMap()
+        .also { it.putAll(agentParams) }
+        .also { it[DRILL_INSTALLATION_DIR_PARAM] = drillInstallationDir }
     logger.info { "result parameters: $resultParams" }
     return resultParams
 }
 
 private fun readFile(filepath: String): String {
-    val isFilename: (String) -> Boolean = { Regex("[\\w\\-.]+").matches(it) }
-    var fileDescriptor = open(filepath, O_RDONLY)
-    if (fileDescriptor == -1 && isFilename(filepath))
```

Comment:
This extra logic is not the responsibility of the readFile function, so I have removed it from here. But I didn't add this logic before the readFile function because it seemed quite strange to me.  I would check if there are actually use cases